src/rdf4cpp/rdf/datatypes/registry/LiteralDatatypeImpl.hpp

```diff
@@ -75,6 +75,14 @@ struct Default {
         str_s << value;
         return str_s.str();
     }
+
+    /**
+     * Returns a string representation of a datatype that is intended to be shown in user friendly output.
+     * E.g. when casting to xsd:string
+     */
+    inline static std::string display(cpp_type const &value) noexcept {
```

Comment:
I don't really get what the difference to to_string is. When are they actually different and what is each function used for? To me it looks right now a bit like there is two solutions for the same problem. 

I had a look at the rust function. If I got it right, it is used as a human-readable counterpart of fmt::Debug. I would say, our to_string is already human readable.

Reply:
The difference is effectively what they are used for.

- `to_string` is used when interacting with the backend as it always yields the canonical representation of a value (which is then stored in the backend, so the representations there are always canonical)
- `display` is used for casting numeric types to `xsd:string` as it does not always require the canonical version but a simplified one. You can basically also use it to output more user friendly representations, which is why I chose that name

Reply:
I would rename:
> to_string -> canonical_string

I am not sure about the name of the second function. How about 
> display -> xsd_string

Reply:
how about:
`to_canonical_string`
`to_simplified_string`
?